lib/input_sanitizer/restricted_hash.rb

```diff
@@ -1,8 +1,33 @@
 module InputSanitizer
   class RestrictedHash < Hash
     def initialize(allowed_keys)
-      @allowed_keys = allowed_keys
-      super() { |hash, key| default_for_key(key) }
+      @allowed_keys = Set.new(allowed_keys)
+      super()
+    end
+
+    def [](key)
```

Comment:
What about `#fetch`? It's a common API so we would like to use it, right?

Reply:
it's also different api that allows you to provide default value and block to evaluate. That's why I didn't add it.